### PR TITLE
raster vector join dialog

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Point-Raster-Join im UI](https://trello.com/c/GynowgJG/370-point-raster-join-im-ui)

File: projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                    type: 'RasterVectorJoin',
+                    params,
+                    vector_sources: [vectorWorkflow.operator],
+                    raster_sources: rasterWorkflows.map(workflow => workflow.operator),
+                }
+            })),
+            mergeMap(workflowId => this.projectService.addLayer(new VectorLayer({
+                workflowId,
+                name: outputLayerName,
+                symbology: vectorLayer.symbology,
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.ts:204]: do we really want to use the same symbology? In our demos we always showed that the new points in a new color now have the raster values attached

File: projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                <input [formControl]="control" matInput type="text">
+                <mat-hint *ngIf="$any($any(form.controls.valueNames).controls[i].errors)?.duplicateName" class="error">
+                    Duplicate Value
+                </mat-hint>
+            </mat-form-field>
+        </div>
+        <section>
+            <label>Aggregation Method:</label>
+            <mat-radio-group formControlName="aggregation" fxLayout="column">
+                <mat-radio-button value="first">Use first value</mat-radio-button>
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.html:22]: I think this needs a different wording and more explanation. But maybe we postpone that to a later point in time where we polish everything?

File: projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<wave-sidenav-header>Extract Raster Values and Add it to a Vector Layer</wave-sidenav-header>
+<form [formGroup]="form" (ngSubmit)="$event.preventDefault();add()" fxLayout="column">
+    <div fxFlex class="container">
+        <wave-layer-selection [types]="allowedVectorTypes" formControlName="vectorLayer"></wave-layer-selection>
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.html:4]: only allow points for now as the operator can't handle lines and polygons

File: projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts [hunk 4]
```diff
@@ -76,11 +77,14 @@ export type OperatorListButtonGroups = Array<{
...
     static readonly DEFAULT_MIXED_OPERATOR_DIALOGS: Array<OperatorListType> = [
-        // {
-        //     component: RasterValueExtractionOperatorComponent,
-        //     type: RasterValueExtractionType,
-        //     description: 'Attach raster values to vector data',
-        // },
+        {
+            component: RasterVectorJoinComponent,
+            type: {
+                NAME: 'Raster Vector Join',
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts:83]: we currently only support Raster point join... at least indicate it in the description

File: projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts [hunk 5]
```diff
@@ -76,11 +77,14 @@ export type OperatorListButtonGroups = Array<{
...
+                ICON_URL: createIconDataUrl('Raster Vector Join'),
+            },
+            description: 'Attach raster values to vector data',
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts:86]: we currently only support points. At least indicate it in the description

**Review by michaelmattig — CHANGES_REQUESTED** [review]
.

File: projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts (hunk 5 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts]: f37ded5

File: projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.ts (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.ts]: c9c2f55

File: projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.html (hunk 2 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.html:39]: a3a3687 (and dd17588 )

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**